Fix shm cleanup tests

stevenengler opened the pull request:
~Fixes a race condition by making the unit tests run single-threaded, and makes sure that there are no pre-existing shm files.~

Fixes a race condition by making the unit tests use temporary directories.

sporksmith reviewed (changes requested):
It's a little unfortunate to force all the rust unit tests to be single-threaded, but probably doesn't make much difference in practice since we don't have that many.

Having the unit  tests mess with the real contents of `/dev/shm` seems fairly undesirable though. e.g. it looks like tests would still fail if there was a still-live shadow simulation running in the background (granted it's a bit unlikely).

It'd probably be better if we added a way to override the root path from `/dev/shm`, and have the tests each create and use their own unique temp dir so that they don't interfere with eachother. Wdyt?

stevenengler commented:
> It's a little unfortunate to force all the rust unit tests to be single-threaded, but probably doesn't make much difference in practice since we don't have that many.
> 
> Having the unit tests mess with the real contents of `/dev/shm` seems fairly undesirable though. e.g. it looks like tests would still fail if there was a still-live shadow simulation running in the background (granted it's a bit unlikely).
> 
> It'd probably be better if we added a way to override the root path from `/dev/shm`, and have the tests each create and use their own unique temp dir so that they don't interfere with eachother. Wdyt?

Changed the PR to use a temporary directory instead.
